### PR TITLE
Raise when trying to cache a through association.

### LINKED ISSUE #220: Raise an exception when trying to cache a relation using :through option
We decided to not support the `:through` option in issue #56, but we don't make this clear to users of the library, leading to bugs that are caught later than necessary.

We should raise an exception when cache_has_one or cache_has_many is used on a relation with the `:through` option to avoid confusion.

### PR DESCRIPTION
**Pull request by dylanahsmith**
@camilo & @arthurnn for review
cc @RichardBlair

Fixes #220

### CODE REVIEW

File: lib/identity_cache/configuration_dsl.rb [hunk 1]
```diff
@@ -98,9 +98,13 @@ def cache_has_many(association, options = {})
...
         deprecate_embed_option(options, false, :ids)
         options[:inverse_name] ||= self.name.underscore.to_sym
-        unless self.reflect_on_association(association)
+
+        unless association_reflection = self.reflect_on_association(association)
           raise AssociationError, "Association named '#{association}' was not found on #{self.class}"
         end
+        if association_reflection.options[:through]
+          raise UnsupportedAssociationError, "caching through associations isn't supported"
+        end
```
Comment (camilo) [inline lib/identity_cache/configuration_dsl.rb]: This code looks super similar to the one in line 145 could be DRYed up with a private method?

**Comment by camilo** [comment]
one comment +1 otherwise

File: lib/identity_cache/configuration_dsl.rb (hunk 1 above)
Comment (dylanahsmith) [inline lib/identity_cache/configuration_dsl.rb]: Yup.  Done.

**Comment by camilo** [comment]
:+1:

**Comment by arthurnn** [comment]
:shirt:

**Comment by byroot** [comment]
:+1: I just stepped on this.

**Comment by dylanahsmith** [comment]
I'll merge this after removing the cached through associations in Shopify.

**Pull request merged**